spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactory.java

```diff
@@ -1295,4 +1316,23 @@ else if (!RabbitUtils.isNormalChannelClose(cause)) {
 
 	}
 
+
+    /**
+     * Name each create thread by this class.
+     */
+    private static class AmqpNamedThreadPoolFactory implements ThreadFactory {
+        private AtomicInteger threadId = new AtomicInteger();
+
```

Comment:
Need 
```java
AmqpNamedThreadPoolFactory() {
    super();
}
```

(package protected - to avoid the compiler creating a synthetic constructor).